nncf/quantization/algorithms/weight_compression/torch_fx_backend.py

```diff
@@ -180,23 +180,13 @@ def set_weight(
         graph: NNCFGraph,
         weight: Tensor,
     ) -> torch.fx.Node:
-        weight_node = graph.get_previous_nodes(node_with_weight)[weight_port_id]
-        graph_node = get_graph_node_by_name(model.graph, weight_node.node_name)
-        if len(graph_node.users) != 1:
-            raise nncf.InternalError(f"Weight Node has {len(graph_node.users)} users, 1 expected.")
-
-        node_with_weight_graph = next(iter(graph_node.users))
-        with model.graph.inserting_before(node_with_weight_graph):
-            new_weight_node = create_getattr_from_value(
-                model, model.graph, node_with_weight.node_name + "_compressed_weight", weight.data
-            )
 
-        args = list(node_with_weight_graph.args)
-        args[weight_port_id] = new_weight_node
-        node_with_weight_graph.args = tuple(args)
-        model.graph.eliminate_dead_code()
-
-        return new_weight_node
+        weight_update_command = FXApplyTransformationCommand(
+            constant_update_transformation_builder(node_with_weight, weight.data)
+        )
+        layout = TransformationLayout()
+        layout.register(weight_update_command)
+        model = FXModelTransformer(model).transform(layout)
```

Comment:
```suggestion
        constant_update_transformation_builder(node_with_weight, weight.data)(model)
```

Reply:
Alright, Looks cleaner. I assumed I have to use transformations to support design decisions.

Reply:
Done!